src/main/java/org/cryptomator/ui/keyloading/hub/RegisterDeviceController.java

```diff
@@ -126,17 +131,19 @@ public void register() {
 					}
 				}).thenApply(user -> {
 					try {
-						assert user.privateKey != null; // api/vaults/{v}/user-tokens/me would have returned 403, if user wasn't fully set up yet
+						assert user.privateKey != null && user.publicKey != null; // api/vaults/{v}/user-tokens/me would have returned 403, if user wasn't fully set up yet
+						var userPublicKey = JWEHelper.decodeECPublicKey(Base64.getDecoder().decode(user.publicKey));
+						migrateLegacyDevices(userPublicKey); // TODO: remove eventually, when most users have migrated to Hub 1.3.x or newer
```

Comment:
The migration is now a "tad" undercover. For a cleaner code, it would be benefical to integrate the migration into the CompletionStage flow, but since the migration is a convienience feature, from my point of view this is ok.

Reply:
I tried but it would be a lot of code duplication, e.g. like this:

```java
.thenApply(user -> {
	try {
		assert user.publicKey != null; // api/vaults/{v}/user-tokens/me would have returned 403, if user wasn't fully set up yet
		var userPublicKey = JWEHelper.decodeECPublicKey(Base64.getDecoder().decode(user.publicKey));
		migrateLegacyDevices(userPublicKey); // TODO: remove eventually, when most users have migrated to Hub 1.3.x or newer
		return user;
	} catch (JWEHelper.KeyDecodeFailedException e) {
		throw new RuntimeException("Server answered with unparsable user key", e);
	}
}).thenApply(user -> {
	try {
		assert user.privateKey != null; // api/vaults/{v}/user-tokens/me would have returned 403, if user wasn't fully set up yet
		var userKey = JWEHelper.decryptUserKey(JWEObject.parse(user.privateKey), setupCodeField.getText());
		return JWEHelper.encryptUserKey(userKey, deviceKeyPair.getPublic());
	} catch (ParseException e) {
		throw new RuntimeException("Server answered with unparsable user key", e);
	}
})
```